src/Features/Core/Portable/QuickInfo/CommonQuickInfoProvider.cs

```diff
@@ -33,21 +27,18 @@ internal abstract class CommonQuickInfoProvider : QuickInfoProvider
         protected virtual bool ShouldCheckPreviousToken(SyntaxToken token)
             => true;
 
-        private async Task<QuickInfoItem?> GetQuickInfoAsync(
-            Document document,
-            SyntaxToken token,
-            int position,
-            CancellationToken cancellationToken)
+        private async Task<QuickInfoItem?> GetQuickInfoCoreAsync(QuickInfoContext context)
         {
+            var token = context.Token;
             if (token != default &&
```

Comment:
🔗 I submitted https://github.com/dotnet/roslyn-analyzers/issues/4682 to address this potential performance issue with a proper analyzer.